From 1963f602426890a81352878ced44279fa4adc3cc Mon Sep 17 00:00:00 2001
From: Bumsik Kim <k.bumsik@gmail.com>
Date: Fri, 1 May 2020 00:13:28 +0900
Subject: [PATCH 3/3] test

---
 scripts/blogsearch_start.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/scripts/blogsearch_start.sh b/scripts/blogsearch_start.sh
index c0f1b7ba1..c9911727f 100755
--- a/scripts/blogsearch_start.sh
+++ b/scripts/blogsearch_start.sh
@@ -10,4 +10,9 @@ apk --no-cache add sqlite-dev
 export JEKYLL_UID=$(id -u)
 export JEKYLL_GID=$(id -g)
 
+id
+ls -la
+ls -la ./jekyll
+pwd
+
 jekyll build -s ./jekyll
-- 
2.26.2

