From fa12c4c4c13680b38a326993f6be333f3d4bf328 Mon Sep 17 00:00:00 2001
From: Bumsik Kim <k.bumsik@gmail.com>
Date: Mon, 27 Apr 2020 15:04:16 +0900
Subject: [PATCH 1/4] add blogsearch in gatsby-config.js

---
 gatsby-config.js | 77 ++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 77 insertions(+)

diff --git a/gatsby-config.js b/gatsby-config.js
index 2637eac9..0d8dcfb7 100644
--- a/gatsby-config.js
+++ b/gatsby-config.js
@@ -17,6 +17,83 @@ module.exports = {
     'MarkdownRemark.frontmatter.author': 'AuthorYaml',
   },
   plugins: [
+    {
+      resolve: 'gatsby-plugin-blogsearch',
+      options: {
+        // Generated blogsearch database file.
+        output: 'reactjs.org.gatsby-example.db.wasm',
+        // Fields configurations
+        fields: {
+          title: {
+            enabled: true,
+            indexed: true,
+            hasContent: true,
+          },
+          body: {
+            enabled: true,
+            indexed: true,
+            hasContent: true,
+          },
+          url: {
+            enabled: true,
+            indexed: false,
+            hasContent: true,
+          },
+          categories: {
+            enabled: true,
+            indexed: true,
+            hasContent: true,
+          },
+          tags: {
+            enabled: false,
+            indexed: true,
+            hasContent: true,
+          },
+        },
+        // GraphQL query used to fetch all data for the search index.
+        query: `
+        {
+          allMarkdownRemark {
+            edges {
+              node {
+                fields {
+                  slug
+                }
+                frontmatter {
+                  title
+                }
+                rawMarkdownBody
+                # excerpt
+                # html
+              }
+            }
+          }
+          site {
+            siteMetadata {
+              siteUrl
+            }
+          }
+        }
+        `,
+        // Function used to map the result from the GraphQL query. This should
+        // return an array of items to index in the form of flat objects
+        // containing field properties to index.
+        normalizer: ({ data }) =>
+          data.allMarkdownRemark.edges.map(({ node }) => {
+            const slug = (node.fields.slug.charAt(0) !== '/' ? '/' : '')  + node.fields.slug;
+            return {
+              title: node.frontmatter.title,
+              body: node.rawMarkdownBody
+                      .replace(/[#`\[\]]+/g, '')
+                      .replace(/\(.*\)/g, '')
+                      .replace(/\s+/g, ' '),
+              url: data.site.siteMetadata.siteUrl + slug,
+              categories: slug !== '/' ? slug.substring(1, slug.indexOf('/', 1)) : '',
+              tags: '',
+            };
+          }),
+      },
+    },
     'gatsby-source-react-error-codes',
     'gatsby-transformer-authors-yaml',
     'gatsby-transformer-home-example-code',
-- 
2.26.2

